pkgs/development/ocaml-modules/eio/default.nix

```diff
@@ -0,0 +1,69 @@
+{ lib
+, stdenv
+, buildDunePackage
+, bigstringaf
+, cstruct
+, dune-configurator
+, fetchurl
+, fmt
+, hmap
+, lwt-dllist
+, mtime
+, optint
+, psq
+, uring
+, alcotest
+, crowbar
+, mdx
+}:
+
+buildDunePackage rec {
+  pname = "eio";
+  version = "0.9";
+
+  minimalOCamlVersion = "5.0";
+  duneVersion = "3";
+
+  src = fetchurl {
+    url = "https://github.com/ocaml-multicore/${pname}/releases/download/v${version}/${pname}-${version}.tbz";
+    sha256 = "z735LkgAYzM/nGqyNO5GrO6tsyCXRMgBzfiidLGZfZk=";
+  };
+
+  propagatedBuildInputs = [
+    bigstringaf
+    cstruct
+    fmt
+    hmap
+    lwt-dllist
+    mtime
+    optint
+    psq
+  ] ++ lib.optionals stdenv.isLinux [
+    uring
```

Comment:
eio doesn't depend on uring, eio_linux does.

Reply:
Oh, do I need to create a series of separate packages then?

Reply:
Removed.

Reply:
right. check https://github.com/nix-ocaml/nix-overlays/tree/master/ocaml/eio for inspiration.